### DISCUSSION TITLE
New module

### DISCUSSION THREAD

**Discussion by SilasK** [metagenome-atlas/atlas#611] [Q&A]
Question from @JoseLopezArcondo 

> Dear Silas,
> 
> I would like to include motility as a "module", is there a way to include this, maybe from your jupyter notebook?
> 
> Best

Do you have a clear idea of what "motility" means? e.g. do you have a list of KO that would make up a module for "motility" ?

If you have a set of KOs (or other annotations) you could try to create your own module. 

I did this for SCFA in this [notebook](https://github.com/trajkovski-lab/CMMG/blob/main/notebooks/Analyze-cold-adapted-microbiota.ipynb). I assume in your case it would be rather a set of genes than a "path".

Eventually, it would also be cool to include it in DRAM. but this would take some time until it's validated. 

An alternative would be https://github.com/hzi-bifo/traitar

**Comment by JoseLopezArcondo** [comment]
Hello Silas, well the fastest one would be I guess I would include all KO functions with the "flagella" word in it here 
https://raw.githubusercontent.com/shafferm/DRAM/master/data/genome_summary_form.tsv
But I am not sure if there are redundant functions there or how to check this, to eventually calculate completeness in a reasonable way...
best
josé

**Reply by JoseLopezArcondo** [reply]
Another question I have is, are modules included in ATLAS from DRAM redundant to some extent with those from DRAM itself? Are product categories from DRAM also present in ATLAS modules?

**Reply by SilasK** [reply]
I use dram 1.4 to annotate genomes. So you get everything from dram with atlas. 
What I do in addition, is to distil/aggregate all kegg modules. I think this is not included in the default dram output. but the definitions are from the dram team..

**Reply by SilasK** [reply]
Here is a example output of dram 1.3 within atlas https://github.com/metagenome-atlas/Tutorial/tree/master/Example/genomes/annotations/dram

**Reply by JoseLopezArcondo** [reply]
Thanks, so results in kegg_modules.tsv should not include some metabolisms in distill/products.tsv then? or there's some possibility of redundance?

**Reply by SilasK** [reply]
The products.tsv from dram is a selection of most important pathways that are also displayed in the html report. I took the same code end run it on all kegg_modules (defined by dram). So there is an overlap.

**Comment by JoseLopezArcondo** [comment]
Thanks, So DRAM products have names and do not mention the module id then, right?  
And are CAZy also included in kegg_modules?

**Reply by SilasK** [reply]
No CAZy are not included in kegg modules
